
## This script is derived from (10.1038/nature25496)
set.seed(20240624)
rm(list = ls())

library(matrixStats)

#### step1: extration and log-transfromation of parental expresison matrix
# extract the expression matrix of parental species
raw.TPM <- read.table("Pekingduck.Muscovyduck.Muleduck.Hinnyduck.AL.OV.TPM.txt", header = T, sep = "\t")
Pek.Mus.raw.TPM <- raw.TPM[,c(1:20, 61:80)]
rownames(Pek.Mus.raw.TPM) <- Pek.Mus.raw.TPM$Geneid
Pek.Mus.raw.TPM <- Pek.Mus.raw.TPM[,-1]

# filter lowly-expressed genes
Pek.Mus.raw.TPM <- Pek.Mus.raw.TPM[apply(Pek.Mus.raw.TPM, 1, min)>=1,]

# log-transformed
Pek.Mus.raw.TPM <- log2(Pek.Mus.raw.TPM + 1)
Pek.Mus.raw.TPM <- as.matrix(Pek.Mus.raw.TPM)

# label sample postion in expression matrix (Pek.Mus.raw.TPM)
Pek.AL = 1:10
Mus.AL = 11:19
Pek.OV = 20:29
Mus.OV = 30:39

# The standard deviation vector for the normal distributions
value_sd1=1
value_sd2=1

# Initializing empty vectors to store the results in the loop
nbgenes=dim(Pek.Mus.raw.TPM)[1]
nbpermut=250

nb_genes_detected=c()
nb_genes_diff_direction=c()

median_per_CDE=c()
Q1_per_CDE=c()
Q3_per_CDE=c()

median_per_CNDE=c()
Q1_per_CNDE=c()
Q3_per_CNDE=c()

# Simple calculation of the 5% cutoff using nbpermut number of points
nb_thresh=floor(0.05*nbpermut/2)


#### step2: define a function to return the t-statistic for all genes at the same time
return_t_stat=function(vect1,vect2){
  rM1=rowMeans(vect1)
  rM2=rowMeans(vect2)
  rV1=rowVars(vect1)
  rV2=rowVars(vect2)
  dim_n1=dim(vect1)[2]
  dim_n2=dim(vect2)[2]
  return((rM1-rM2) / sqrt(rV1/dim_n1 + rV2/dim_n2))
}


### step3: Create "nbpermut" permuted data sets; nbpermut=250
# Permuted data sets were generated by randomly reassigning entire RNA-seq samples among population and treatment
# permutatations are performed separately for each comparison
permutations = matrix(rep(0, 39*nbpermut), nrow=nbpermut)
for(i in 1:nbpermut) {permutations[i,] <- c(sample(c(1:39), 39))}

dist_Pek_Mus_OV = matrix(rep(0, nbgenes*nbpermut), nrow=nbgenes)
dist_Pek_Mus_AL = matrix(rep(0, nbgenes*nbpermut), nrow=nbgenes)

dist_Pek_AL_Mus_OV = matrix(rep(0, nbgenes*nbpermut), nrow=nbgenes)
dist_Pek_OV_Mus_AL = matrix(rep(0, nbgenes*nbpermut), nrow=nbgenes)

# And for the ancestral plasticity
dist_Pek_AL.OV = matrix(rep(0, nbgenes*nbpermut), nrow=nbgenes)

# Calculate the t-statistic distribution for each gene and comparison
for(per in 1:nbpermut){
  all_counts_per = Pek.Mus.raw.TPM[, permutations[per,]]
  # Pek and Mus
  dist_Pek_Mus_OV[, per] = return_t_stat(all_counts_per[, Pek.OV], all_counts_per[, Mus.OV])
  dist_Pek_Mus_AL[, per] = return_t_stat(all_counts_per[, Pek.AL], all_counts_per[, Mus.AL])

  dist_Pek_AL_Mus_OV[, per] = return_t_stat(all_counts_per[, Pek.AL], all_counts_per[, Mus.OV])
  dist_Pek_OV_Mus_AL[, per] = return_t_stat(all_counts_per[, Pek.OV], all_counts_per[, Mus.AL])  
  
  # Ancestral plasticity
  dist_Pek_AL.OV[,per] =  return_t_stat(all_counts_per[, Pek.AL],  all_counts_per[, Pek.OV])
}

## Sort the distributions
dist_Pek_Mus_OV_s = t(apply(dist_Pek_Mus_OV, 1, sort))
dist_Pek_Mus_AL_s = t(apply(dist_Pek_Mus_AL, 1, sort))

dist_Pek_AL_Mus_OV_s = t(apply(dist_Pek_AL_Mus_OV, 1, sort))
dist_Pek_OV_Mus_AL_s = t(apply(dist_Pek_OV_Mus_AL, 1, sort))

dist_Pek_AL_OV_s  = t(apply(dist_Pek_AL.OV,  1, sort))


### step4: Test for differential expression
# Get the t-statictic for every comparison and check if it falls in 5% tail of permuted t-statistic distribution
diff_DE_Pek_Mus_OV = return_t_stat(Pek.Mus.raw.TPM[, Pek.OV], Pek.Mus.raw.TPM[, Mus.OV])
is_DE_Pek_Mus_OV   = diff_DE_Pek_Mus_OV < dist_Pek_Mus_OV_s[, c(nb_thresh+1)] | diff_DE_Pek_Mus_OV > dist_Pek_Mus_OV_s[, c(nbpermut-nb_thresh)]

diff_DE_Pek_Mus_AL = return_t_stat(Pek.Mus.raw.TPM[, Pek.AL], Pek.Mus.raw.TPM[, Mus.AL])
is_DE_Pek_Mus_AL   = diff_DE_Pek_Mus_OV < dist_Pek_Mus_AL_s[, c(nb_thresh+1)] | diff_DE_Pek_Mus_AL > dist_Pek_Mus_AL_s[, c(nbpermut-nb_thresh)]

### step5: Test for plasticity in Peking and Muscovy ducks population
diff_DE_Pek_AL_OV = return_t_stat(Pek.Mus.raw.TPM[, Pek.AL], Pek.Mus.raw.TPM[, Pek.OV])
is_DE_Pek_AL_OV   = diff_DE_Pek_AL_OV < dist_Pek_AL_OV_s[, c(nb_thresh+1)] | diff_DE_Pek_AL_OV > dist_Pek_AL_OV_s[, c(nbpermut-nb_thresh)]

diff_DE_Pek_AL_Mus_OV = return_t_stat(Pek.Mus.raw.TPM[, Pek.AL], Pek.Mus.raw.TPM[, Mus.OV])
is_DE_Pek_AL_Mus_OV   = diff_DE_Pek_AL_Mus_OV < dist_Pek_AL_Mus_OV_s[, c(nb_thresh+1)] | diff_DE_Pek_AL_Mus_OV > dist_Pek_AL_Mus_OV_s[, c(nbpermut-nb_thresh)]

diff_DE_Pek_OV_Mus_AL = return_t_stat(Pek.Mus.raw.TPM[, Pek.OV], Pek.Mus.raw.TPM[, Mus.AL])
is_DE_Pek_OV_Mus_AL   = diff_DE_Pek_AL_Mus_OV < dist_Pek_OV_Mus_AL_s[, c(nb_thresh+1)] | diff_DE_Pek_AL_Mus_OV > dist_Pek_AL_Mus_OV_s[, c(nbpermut-nb_thresh)]


### step6: Get the evolved change in Muscovy duck populations
diff_DE_Mus_AL_OV = return_t_stat(Pek.Mus.raw.TPM[, Mus.AL], Pek.Mus.raw.TPM[, Mus.OV])


### step7: Apply conditioning to test for directionality
# Test concordantly differentially expressed (CDE) or not (CDNE)
is_CDE = (is_DE_Pek_Mus_OV & is_DE_Pek_Mus_AL & is_DE_Pek_AL_Mus_OV & is_DE_Pek_OV_Mus_AL)
nb_genes_detected = c(nb_genes_detected, sum(is_CDE))
sum(is_CDE)

is_CNDE= !(is_DE_Pek_Mus_OV & is_DE_Pek_Mus_AL & is_DE_Pek_AL_Mus_OV & is_DE_Pek_OV_Mus_AL)
nb_genes_diff_direction = c(nb_genes_diff_direction, sum(is_CNDE))


### step8: calculate direction of ancestral plasticity and evolved plasticity 
## Mean change between ancestral plasticity and evolved plasticity
diff_Means = rowMeans(Pek.Mus.raw.TPM[,Mus.OV]) - rowMeans(Pek.Mus.raw.TPM[,Pek.OV])

# Ancestral plasticity
diff_anc_plast = rowMeans(Pek.Mus.raw.TPM[,Pek.OV]) - rowMeans(Pek.Mus.raw.TPM[,Pek.AL])

# Direction of ancestral and evolved plasticity
dir_evol_CDE  = sign(diff_DE_Pek_Mus_OV)[is_CDE]
dir_plast_CDE = sign(diff_anc_plast)[is_CDE]

# Then the chisquare likelihood on the CDE transcripts
Chi_stat = chisq.test(as.factor(dir_evol_CDE), as.factor(dir_plast_CDE))$stat


####  step10: Now CDE and CNDE in the permuted data sets
## We run the same procedure on each of the permutation
nb_genes_inper=c()
nb_genes_CNDE_inper=c()
dist_Chi_stat_per=c()

for(kkk in 1:nbpermut){
  all_counts_per= Pek.Mus.raw.TPM[, permutations[kkk,]]
  
  diff_DE_Pek_Mus_OV_per = return_t_stat(all_counts_per[,Pek.OV], all_counts_per[, Mus.OV])
  is_DE_Pek_Mus_OV_per = diff_DE_Pek_Mus_OV_per < dist_Pek_Mus_OV_s[,c(nb_thresh+1)] | diff_DE_Pek_Mus_OV_per > dist_Pek_Mus_OV_s[,c(nbpermut-nb_thresh)]
  
  diff_DE_Pek_Mus_AL_per = return_t_stat(all_counts_per[,Pek.AL], all_counts_per[, Mus.AL])
  is_DE_Pek_Mus_AL_per = diff_DE_Pek_Mus_AL_per < dist_Pek_Mus_AL_s[,c(nb_thresh+1)] | diff_DE_Pek_Mus_AL_per > dist_Pek_Mus_OV_s[,c(nbpermut-nb_thresh)]
  
  diff_DE_Pek_OV_Mus_AL_per = return_t_stat(all_counts_per[,Pek.OV], all_counts_per[, Mus.AL])
  is_DE_Pek_OV_Mus_AL_per = diff_DE_Pek_OV_Mus_AL_per < dist_Pek_OV_Mus_AL_s[,c(nb_thresh+1)] | diff_DE_Pek_OV_Mus_AL_per > dist_Pek_OV_Mus_AL_s[,c(nbpermut-nb_thresh)]
  
  diff_DE_Pek_AL_Mus_OV_per = return_t_stat(all_counts_per[,Pek.AL], all_counts_per[, Mus.OV])
  is_DE_Pek_AL_Mus_OV_per = diff_DE_Pek_AL_Mus_OV_per < dist_Pek_AL_Mus_OV_s[,c(nb_thresh+1)] | diff_DE_Pek_AL_Mus_OV_per > dist_Pek_OV_Mus_AL_s[,c(nbpermut-nb_thresh)]
  
  is_CDE_per = (is_DE_Pek_Mus_OV_per & is_DE_Pek_Mus_AL_per & is_DE_Pek_AL_Mus_OV_per & is_DE_Pek_OV_Mus_AL_per)
  nb_genes_inper = c(nb_genes_inper, sum(is_CDE_per))
  
  is_CNDE_per= !(is_DE_Pek_Mus_OV_per & is_DE_Pek_Mus_AL_per & is_DE_Pek_AL_Mus_OV_per & is_DE_Pek_OV_Mus_AL_per)
  nb_genes_CNDE_inper=c(nb_genes_CNDE_inper, sum(is_CNDE_per))
  
  # And we can add here the chisq statistics for the association test
  diff_anc_plast_per = rowMeans(Pek.Mus.raw.TPM[, Pek.OV]) - rowMeans(Pek.Mus.raw.TPM[, Pek.AL])
  dir_evol_CDE_per   = sign(diff_DE_Pek_Mus_OV_per)[is_CDE]
  dir_plast_CDE_per  = sign(diff_anc_plast_per)[is_CDE]
  
  # Then the chisquare likelihood on the CDE genes
  dist_Chi_stat_per=c(dist_Chi_stat_per,chisq.test(as.factor(dir_evol_CDE_per), as.factor(dir_plast_CDE_per))$stat)
}

# calculate Q1 and Q2 values of CDEs detected in permuted data sets
median_per_CDE=c(median_per_CDE, summary(nb_genes_inper)[3])
Q1_per_CDE=c(Q1_per_CDE, summary(nb_genes_inper)[2])
Q3_per_CDE=c(Q3_per_CDE, summary(nb_genes_inper)[5])

# calculate Q1 and Q2 values of CNDEs detected in permuted data sets
median_per_CNDE=c(median_per_CNDE, summary(nb_genes_CNDE_inper)[3])
Q1_per_CNDE=c(Q1_per_CNDE, summary(nb_genes_CNDE_inper)[2])
Q3_per_CNDE=c(Q3_per_CNDE, summary(nb_genes_CNDE_inper)[5])

sum(is_CDE)
sum(is_CNDE)

sum(is_CDE_per)
sum(is_CNDE_per)

summary(nb_genes_inper)
summary(nb_genes_CNDE_inper)


### step11: Permute the population means 1000 times, keeping only the CDE genes
all_means = cbind(rowMeans(Pek.Mus.raw.TPM[,Pek.AL]), rowMeans(Pek.Mus.raw.TPM[,Pek.OV]), rowMeans(Pek.Mus.raw.TPM[,Mus.AL]), rowMeans(Pek.Mus.raw.TPM[,Mus.OV]))
all_means = all_means[is_CDE,]

vect_corr=c()
vect_corr2=c()

for(kkk in 1:1000){
  all_per=all_means
  for(j in 1:sum(is_CDE)){
    all_per[j,] = all_means[j, sample(1:4, 4)]
  }
  # Now we compute the distributions plotted in the two insets
  diff_Means_per = - (all_per[,2] - all_per[,4])
  diff_anc_plast_per = all_per[,2] - all_per[,1]
  new_plast_per = all_per[,4] - all_per[,3]
  
  # Genes that change plasticity
  change_plast_per = new_plast_per - diff_anc_plast_per
  
  vect_corr  = c(vect_corr,  cor(diff_Means_per,   diff_anc_plast_per, method="spearman"))
  vect_corr2 = c(vect_corr2, cor(change_plast_per, diff_anc_plast_per, method="spearman"))
}


### step12: Plotting the main figure
library(ggplot2)
library(grid)

## plot1
plot.dat <- data.frame(diff_means = diff_Means[is_CDE], diff_anc_plast = diff_anc_plast[is_CDE])
plot.dat$correlation <- "negative"
plot.dat$correlation[plot.dat$diff_means*plot.dat$diff_anc_plast > 0] <- "positive"

p1 <- ggplot(plot.dat, aes(x=diff_anc_plast, y=diff_means, color=correlation)) + geom_point()
p2 <- p1 + geom_smooth(data = subset(plot.dat, correlation == "positive"), method = "lm", se = TRUE, color = "blue", size=0.5)
p3 <- p2 + geom_vline(xintercept = 0, linetype = 4, size = 0.5) + geom_hline(yintercept = 0, linetype = 4, size = 0.5)
p4 <- p3 + scale_x_continuous(limits = c(-5.5, 5.5), breaks = c(-5,-4,-3,-2,-1,0,1,2,3,4,5), expand = c(0,0)) + 
           scale_y_continuous(limits = c(-5.5, 6.5), breaks = c(-5,-4,-3,-2,-1,0,1,2,3,4,5,6), expand = c(0,0)) + coord_fixed()
p5 <- p4 + scale_colour_manual(values = c("#DA3C3D", "#4289C2"), limits = c("positive","negative")) + guides(colour = "none")
p6 <- p5 + labs(x = "Ancestral plasticity in gene expression\n(log-transformed TPMs)", y = "Evolved divergence in gene expression\n(log-transformed TPMs)") + 
           theme(panel.background = element_rect(fill="white"))
p7 <- p6 + theme(axis.line = element_line(colour="black", size = 1),
                 axis.title.x = element_text(size = 13, vjust = -2), axis.title.y = element_text(size = 13,vjust = 3),
                 axis.text = element_text(colour="black", size = 10))
fig1 <- p7 + theme(plot.margin = unit(c(0.2,0.2,0.2,0.2),"inches"))

pdf(file = "adaptive.plasticy.correlation.pdf", height = 4, width = 4)
fig1
dev.off()

## plot2
plot1.dat <- data.frame(coefficient=vect_corr)
p1 <- ggplot(plot1.dat, aes(x=coefficient)) + geom_histogram(bins = 200, binwidth = 0.002)
p2 <- p1 + scale_x_continuous(limits = c(-0.6, 0.25), expand = c(0,0)) + scale_y_continuous(expand = c(0,0)) +
      labs(x="correlation coefficient", y="count") + theme(panel.background = element_rect(fill="white"))
p3 <- p2 + theme(axis.line = element_line(colour="black", size = 0.5),
                 axis.title.x = element_text(size = 12, vjust = -2), axis.title.y = element_text(size = 12, vjust = 3),
                 axis.text = element_text(colour="black", size = 10))
p4 <- p3 + theme(plot.margin = unit(c(0.2,0.2,0.2,0.2),"inches"))
p5 <- p4 + annotate("segment", x=cor(diff_Means[is_CDE], diff_anc_plast[is_CDE]),
                      xend = cor(diff_Means[is_CDE], diff_anc_plast[is_CDE]), y = 28, yend = 0.8, colour="red", size=0.7, arrow=arrow(length = unit(0.3, "cm")))
fig2 <- p5 + annotate("text", x=0.17, y=32, label="r=0.198")

pdf(file = "adaptive.plasticy.correlation.simualtion.pdf", height = 2, width = 4)
fig2
dev.off()


#### step13: save results
# genes with adaptive plasticity
adap.genes <- subset(plot.dat, correlation=="positive")
write.csv(adap.genes, "adpative.plasticity.genes.csv", row.names = T, quote = F)

# genes with non-adaptive plasticity
non.adap.genes <- subset(plot.dat, correlation=="negative")
write.csv(non.adap.genes, "nonadpative.plasticity.genes.csv", row.names = T, quote = F)
